### PR TITLE
Python 3 compatibility: Use renamed urllib modules

### PR DESCRIPTION
**Pull request by saschanaz**
Some urllib functions are moved on Python 3 so this PR uses the renamed ones.

Bonus: This PR uses [assertCountEqual which is renamed from assertItemsEqual](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertItemsEqual).

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -28,13 +28,18 @@
...
 
-import os, sys, shutil, tempfile, subprocess, shlex, time, re, logging, urllib
+import os, sys, shutil, tempfile, subprocess, shlex, time, re, logging
 from subprocess import PIPE
 from tools import shared, jsrun, system_libs
 from tools.shared import execute, suffix, unsuffixed, unsuffixed_basename, WINDOWS, safe_move
 from tools.response_file import read_response_file
 import tools.line_endings
 
+try:
```
Comment (kripken) [inline emcc.py]: maybe here and in the other places add a comment saying this is for 2/3 support? that way when we can remove 2 support we could search for such things and remove the 2 parts. also it looks weird by itself, not sure it's obvious it's for 2/3 compat.

Comment (saschanaz) [reply emcc.py]: That will be great, done.

(Will we be able to remove Py2 support in 2025? I hope so.)

File: tests/test_other.py [hunk 2]
```diff
@@ -7047,7 +7047,10 @@ def test_link_response_file_does_not_force_absolute_paths(self):
     shutil.rmtree(os.path.join(current_directory, with_space))
 
     # We want only the relative path to be in the linker args, it should not be converted to an absolute path.
-    self.assertItemsEqual(link_args, [main_object_file_name])
+    if hasattr(self, 'assertCountEqual'):
```
Comment (kripken) [inline tests/test_other.py]: maybe also mention python 2 compat on this one? (again, it looks odd without an explanation)

Comment (saschanaz) [reply tests/test_other.py]: Oops, sorry. Also done.

**Pull request merged**